mgmt-lambda/app.ts

```diff
@@ -1,175 +1,66 @@
-    IfMatch: cfConfig.ETag,
+  try {
+    const authSettings = await getAuthSettings(secretManagerClient)
+    const authorization = event.headers['authorization']
+    if (authorization !== authSettings.token) {
+      return handleNoAthentication()
+    }
+  } catch (error) {
+    return handleWrongConfiguration(error)
   }
 
-  const updateConfigCommand = new UpdateDistributionCommand(updateParams)
-  const updateCFResult = await cloudFrontClient.send(updateConfigCommand)
-  console.info(`CloudFront update has finished, ${JSON.stringify(updateCFResult)}`)
-
-  console.info('Going to invalidate routes for upgraded cache behavior')
-  if (!cacheBehavior.PathPattern) {
-    return publishJobFailure(ctx, job, 'Path pattern is not defined')
+  let deploymentSettings: DeploymentSettings
+  try {
+    deploymentSettings = loadDeploymentSettings()
+  } catch (error) {
+    return handleWrongConfiguration(error)
   }
 
-  let pathPattern = cacheBehavior.PathPattern
-  if (!pathPattern.startsWith('/')) {
-    pathPattern = '/' + pathPattern
-  }
+  const path = event.rawPath
+  const method = event.requestContext.http.method
+  const lambdaClient = new LambdaClient({ region: defaults.AWS_REGION })
+  const cloudFrontClient = new CloudFrontClient({ region: defaults.AWS_REGION })
 
-  const invalidationParams: CreateInvalidationCommandInput = {
-    DistributionId: cloudFrontDistrId,
-    InvalidationBatch: {
-      Paths: {
-        Quantity: 1,
-        Items: [pathPattern],
-      },
-      CallerReference: 'fingerprint-pro-management-lambda-function',
-    },
+  if (path.startsWith('/update') && method === 'POST') {
+    return handleUpdate(lambdaClient, cloudFrontClient, deploymentSettings)
+  } else if (path.startsWith('/status') && method === 'GET') {
+    return handleStatus(lambdaClient, deploymentSettings)
+  } else {
+    return handleNotFound()
   }
-  const invalidationCommand = new CreateInvalidationCommand(invalidationParams)
-  const invalidationResult = await cloudFrontClient.send(invalidationCommand)
-  console.info(`Invalidation has finished, ${JSON.stringify(invalidationResult)}`)
-
-  await publishJobSuccess(ctx, job)
 }
 
-async function getLambdaLatestVersionArn(functionName: string): Promise<string | undefined> {
-  const client = new LambdaClient({ region: REGION })
-  const params: ListVersionsByFunctionCommandInput = {
-    FunctionName: functionName,
+function loadDeploymentSettings(): DeploymentSettings {
+  const cfDistributionId = process.env.CFDistributionId
+  if (!cfDistributionId) {
+    throw new Error('No CloudFront distribution Id')
   }
-  const command = new ListVersionsByFunctionCommand(params)
-  const result: ListVersionsByFunctionCommandOutput = await client.send(command)
-  if (!result.Versions || result.Versions?.length === 0) {
-    return Promise.resolve(undefined)
+  const lambdaFunctionName = process.env.LambdaFunctionArn
+  if (!lambdaFunctionName) {
+    throw new Error('No lambda function name')
```

Comment:
I'd suggest to have more clear error message, for instance `load failed: env variable LambdaFunctionArn not found`.

Reply:
This seems to be duplicated. Did you mean something like `process.env.LambdaFunctionName`?